lib/app.rb

```diff
@@ -10,145 +10,139 @@ class App
   def initialize(output_stream, input_stream)
     @output_stream = output_stream
     @input_stream = input_stream
+    @projects = []
   end
 
   def run
-    @output_stream.puts("\e[38;5;40mWelcome to Taskitome!")
-    @output_stream.puts("\e[0;37m=============================\n")
-    @output_stream.puts("\e[0mPROJECTS MENU")
-    @output_stream.puts("\e[0;37m-----------------------------")
-    @output_stream.puts("\e[40;38;5;214mENTER A COMMAND:\e[0m")
-    @output_stream.puts("\e[1;37ma   \e[0;35mAdd a new project")
-    @output_stream.puts("\e[1;37mls  \e[0;35mList all project")
-    @output_stream.puts("\e[1;37md   \e[0;35mDelete a project")
-    @output_stream.puts("\e[1;37me   \e[0;35mEdit a project")
-    @output_stream.puts("\e[1;37mq   \e[0;35mQuit the app\e[0m\n\n")
+    output_stream.puts("\e[38;5;40mWelcome to Taskitome!")
+    output_stream.puts("\e[0;37m=============================\n")
+    output_stream.puts("\e[0mPROJECTS MENU")
+    output_stream.puts("\e[0;37m-----------------------------")
+    output_stream.puts("\e[40;38;5;214mENTER A COMMAND:\e[0m")
+    output_stream.puts("\e[1;37ma   \e[0;35mAdd a new project")
+    output_stream.puts("\e[1;37mls  \e[0;35mList all project")
+    output_stream.puts("\e[1;37md   \e[0;35mDelete a project")
+    output_stream.puts("\e[1;37me   \e[0;35mEdit a project")
+    output_stream.puts("\e[1;37mq   \e[0;35mQuit the app\e[0m\n\n")
 
     capture_command
 
     while command != "q"
-      if !@current_project
+      if no_current_project?
         case command
         when "a"
-          @projects = [] if @projects.nil?
-          @output_stream.puts("\e[0;3mEnter a project name:\e[0m")
-          name = @input_stream.gets.chomp
-          @projects << {name => []}
-          @output_stream.puts("\e[38;5;40mCreated project:\e[0m '#{name}'\n\n")
+          capture_project_name
+          projects << {project_name => []}
+          output_stream.puts("\e[38;5;40mCreated project:\e[0m '#{project_name}'\n\n")
```

Comment:
I'm feeling the same way about `@output_stream` extraction. Seems like `@output_stream.puts` to `print` would be a better path through this morass.